app/controllers/messages_controller.rb

```diff
@@ -44,14 +45,14 @@ def destroy
   private
 
   def set_conversation
-    @conversation = Conversation.find(params[:conversation_id])
+    @conversation = Current.user.conversations.find(params[:conversation_id])
   end
 
   def set_message
     @message = Message.find(params[:id])
   end
 
   def message_params
-    params.require(:message).permit(:role, :content_text)
+    params.require(:message).permit(:content_text)
```

Comment:
But message does need the role attribute set, right? I thought maybe you modified message.rb to have a default value but I didn't see anything like that. Won't role end up being nil for all newly created messages?

Reply:
The message gets the role set [in the controller action](https://github.com/hostedgpt/hostedgpt/pull/58/files/2b2c4ec716b6452c20bc69862347996ea6c8cccc#diff-9b58332ea76756301b1aafeacfb0659d1244ba014947602b0e0bf2d0f2921d26R22), not in the message params. Allowing it to be set in the message params means someone could "impersonate" the assistant by putting `role=assistant` into the form data manually.

In this case it's probably not a security concern, but the notion of setting the role in this controller is implicitly to set it to "user". I favor making that explicit in code rather than submitting a parameter in the form.